src/runtime/hexagon_host.cpp

```diff
@@ -112,7 +112,13 @@ WEAK int init_hexagon_runtime(void *user_context) {
     // code execution based on the runtime; devices with Hexagon hardware will
     // simply provide conduits for execution on that hardware, while test/desktop/etc
     // environments can instead connect a simulator via the API.
+    // If we are a windows hosted simulator, then we need to load the dll file.
+    // Otherwise if we are a linux hosted simulator we need the .so.
```

Comment:
It's really that if we aren't on the windows simulator, then we want the .so, whether it's linux or android (on device).